includes/admin/class-omise-page-settings.php

```diff
@@ -27,6 +27,10 @@ protected function save( $data ) {
 			$data['account_email']   = $account['email'];
 			$data['account_country'] = $account['country'];
 
+			$capabilities = Omise_Capabilities::retrieve( $public_key, $secret_key );
+			$backends = $capabilities->getBackends();
```

Comment:
This will lead to error when the keys are empty or wrong. Value of `$capabilities` will be `null` so `$capabilities->getBackends()` will throw `Call to a member function getBackend() on null` error.

Please add necessary checks to avoid the error.

Reply:
but are not we in the try catch section 🤔 
```php
		try {
			$account = OmiseAccount::retrieve( $public_key, $secret_key );

			$data['account_id']      = $account['id'];
			$data['account_email']   = $account['email'];
			$data['account_country'] = $account['country'];

			$capabilities = Omise_Capabilities::retrieve( $public_key, $secret_key );
			$backends = $capabilities->getBackends();
			$data['backends'] = $backends;

			$this->update_settings( $data );
			$this->add_message(
				'message',
				sprintf( __( 'The settings have been saved, an account: %s has been connected.', 'omise' ), $account['email'] )
			);
		} catch (Exception $e) {
			$this->add_message( 'error', $e->getMessage() );
		}
```
so if got exceptions from capabilities is should got catch and display the error message (on my testing is also help us validate key before save)

Reply:
Ah sorry I didn't look the whole code block. Thanks for clearing my doubt.